src/FA2.hs

```diff
@@ -0,0 +1,122 @@
+module FA2 
+  ( tokenMetadataRegistry
+  , updateOperator
+  ) where
+
+import Text.Show (Show (..))
+
+import Lorentz
+import Lorentz.Contracts.Spec.FA2Interface hiding (name, sender)
+
+type Operators = BigMap (Address, Address) ()
+
+type Ledger = BigMap (Address, TokenId) Natural
+
+type StorageTokenMetadata = ("token_metadata" :! BigMap TokenId TokenMetadata)
+
+-- | Storage of the FA2 contract
+data Storage = Storage
+  { sLedger      :: Ledger
+  , sOperators   :: Operators
+  , sTokenMetadata :: StorageTokenMetadata
+  }
+  deriving stock (Generic, Show)
+  deriving anyclass (IsoValue, HasAnnotation)
+
+instance TypeHasDoc Storage where
+  typeDocMdDescription =
+    "Contract's storage holding a big_map with all balances and the operators."
+
+tokenMetadataRegistry :: forall store. Entrypoint TokenMetadataRegistryParam store
+tokenMetadataRegistry = do
+  sender
+  push @Mutez (toMutez 0)
+  swap
+  transferTokens # nil # swap # cons # pair
+
+updateOperator :: Entrypoint UpdateOperatorsParam Storage
+updateOperator = do
+  iter $
+    caseT @UpdateOperator
+      ( #cAdd_operator /-> addOperator
+      , #cRemove_operator /-> removeOperator
+      )
+  nil; pair    
+
+addOperator :: forall s. OperatorParam & Storage & s :-> Storage & s
+addOperator = do
+  unpair
+  fromNamed #owner
+  swap
+  unpair
+  fromNamed #operator
+  swap
+  drop @("token_id" :! TokenId)
+  swap
+  dupTop2
+  assertNeq [mt|NOT_OWNER|]
+  dig @2
+  getField #sOperators
+  dig @2
+  dig @3
+  pair
+  dupTop2
+  mem
+  if_ ifKeyExists ifKeyDoesntExist
+  where
+    dupTop2 :: a & b & f :-> a & b & a & b & f
+    dupTop2 = do
+      duupX @2
+      duupX @2
+
+    ifKeyExists :: (Address, Address) & Operators & Storage & s :-> Storage & s
+    ifKeyExists = dropN @2
+
+    ifKeyDoesntExist :: (Address, Address) & Operators & Storage & s :-> Storage & s
+    ifKeyDoesntExist = do
+      unit
+      some
+      swap
+      update
+      setField #sOperators
+
+
+removeOperator :: forall s. OperatorParam & Storage & s :-> Storage & s
+removeOperator = do
+  unpair
+  fromNamed #owner
+  swap
+  unpair
+  fromNamed #operator
+  swap
+  drop @("token_id" :! TokenId)
+  swap
+  dupTop2
+  assertNeq [mt|NOT_OWNER|]
+  dig @2
+  getField #sOperators
+  dig @2
+  dig @3
+  pair
+  dupTop2
+  mem
+  if_ ifKeyExists ifKeyDoesntExist
+  where
+    dupTop2 :: a & b & f :-> a & b & a & b & f
```

Comment:
Please, don't copy-paste it. In fact, I think it makes sense to add this function to Lorentz, will make an issue probably.